src/DotVVM.Framework/Controls/Internal.cs

```diff
@@ -60,6 +61,11 @@ public class Internal
 
         public static DotvvmProperty ReferencedViewModuleInfoProperty =
             DotvvmProperty.Register<ViewModuleReferenceInfo, Internal>(() => ReferencedViewModuleInfoProperty, isValueInherited: true);
+
+        public static bool IsViewCompilerProperty(DotvvmProperty property)
+        {
+            return property == DataContextTypeProperty;
```

Comment:
```suggestion
            return property.DeclaringType == typeof(Internal);
```

There are other properties (like unique id, markup line, ..) which might be there.